mantidproject/mantid
# Move file loading and processing async in SANS

**Description of work.**
Move file finding to be synchronous as we need the result before
processing or loading data. Instead we now run the load or process
method async within the worker to avoid blocking the main thread

**To test:**
- Using the LOQ data set:
- Attempt to load or process a data set
- An error previously would appear and pressing ok and process again would work
- Now it should process first time
- Try various combo's with load and process selected/all

Fixes #27216 .

*This does not require release notes* because this issue was introduced and resolved before a release

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
